samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -401,6 +404,34 @@ static int burst_write(u8_t reg, const u8_t *buf, size_t size)
 	return err;
 }
 
+static void update_cpi(u16_t cpi)
+{
+	/* Set resolution with CPI step of 100 cpi
+	 * 0x00: 100 cpi (minimum cpi)
+	 * 0x01: 200 cpi
+	 * :
+	 * 0x31: 5000 cpi (default cpi)
+	 * :
+	 * 0x77: 12000 cpi (maximum cpi)
+	 */
+
+	if (cpi > OPTICAL_MAX_CPI || cpi < OPTICAL_MIN_CPI) {
+		LOG_WRN("CPI value out of range");
+		return;
+	}
+
+	/* Convert CPI to register value */
+	uint8_t value = (cpi / 100) - 1;
+
+	LOG_INF("Setting CPI to %u (reg value 0x%x)", cpi, value);
+
+	int err = reg_write(OPTICAL_REG_CONFIG1, value);
```

Comment:
`reg_write` cannot be called from this context as it can get in the middle of communication performed by sensore reading thread
We could do the following: set the `atomic_t cpi;` and in the thread keep a local value of cpi - when cpi value differs from local you set the register.